pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GatePragmas.hpp

```diff
@@ -0,0 +1,36 @@
+// Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @file GatePragmas.hpp
+ * Defines macros for enabling various OpenMP options in the gate kernel
+ * definitions.
+ */
+#pragma once
+
+namespace Pennylane::LightningQubit::Gates::Pragmas {
+
+// Defines utility macros to annotate gate-kernel loops with OpenMP parallel-for
+// and OpenMP SIMD pragmas. Selectable at compile time.
+#if defined PL_LQ_KERNEL_OMP && defined _OPENMP
+#define PRAGMA_WRAP(S) _Pragma(#S)
+#define PL_LOOP_PARALLEL(x) PRAGMA_WRAP(omp parallel for collapse(x))
+#define PL_LOOP_PARALLEL_VA(x, ...) PRAGMA_WRAP(omp parallel for collapse(x) __VA_ARGS__)
+#define PL_LOOP_SIMD PRAGMA_WRAP(omp simd)
```

Comment:
Where do these SIMD loops become beneficial compared to parallel_for loops for updating indices?

Reply:
We had issues in the past with lackluster OMP support, and failings due to clang/msvc/etc where the macro expanded but failed at the clang tidy layer. This definition rule allow us to simply turn everything off with the -D flag we want, or tune/expand these to suit needs. As we will be doing more ARM stuff this year, I suspect we may need selective compilation based on needs.

Additionally, since we have OpenMP at two layers of LQ, this allows us to turn off the gate-level layer as above.